api/src/test/java/com/redhat/insights/tls/PEMSupportTest.java

```diff
@@ -1,24 +1,82 @@
-        .createTLSContext(
-            getPathFromResource("com/redhat/insights/tls/dummy.cert"),
-            getPathFromResource("com/redhat/insights/tls/dummy.key"));
+  public void testLoadingCertAndKey()
+      throws NoSuchAlgorithmException, CertificateException, IOException {
+    SelfSignedX509CertificateAndSigningKey ca =
+        SelfSignedX509CertificateAndSigningKey.builder()
+            .setDn(
+                new X500Principal(
+                    "O=Root Certificate Authority, EMAILADDRESS=elytron@wildfly.org, C=UK,"
+                        + " ST=Elytron, CN=Elytron CA"))
+            .setKeyAlgorithmName("RSA")
+            .setSignatureAlgorithmName("SHA256withRSA")
+            .addExtension(false, "BasicConstraints", "CA:true,pathlen:2147483647")
+            .build();
+    KeyPairGenerator keyPairGenerator = KeyPairGenerator.getInstance("RSA");
+    KeyPair generatedKeys = keyPairGenerator.generateKeyPair();
+    PublicKey publicKey = generatedKeys.getPublic();
+    X509Certificate subjectCertificate =
+        new X509CertificateBuilder()
+            .setIssuerDn(ca.getSelfSignedCertificate().getIssuerX500Principal())
+            .setSubjectDn(new X500Principal("O=Elytron, OU=Elytron, C=UK, ST=Elytron, CN=Firefly"))
+            .setSignatureAlgorithmName("SHA256withRSA")
+            .setSigningKey(ca.getSigningKey())
+            .setNotValidBefore(ZonedDateTime.now().minusYears(1))
+            .setNotValidAfter(ZonedDateTime.now().plusYears(1))
+            .setPublicKey(publicKey)
+            .build();
+    ByteStringBuilder target = new ByteStringBuilder();
+    Pem.generatePemX509Certificate(target, ca.getSelfSignedCertificate());
+    Pem.generatePemX509Certificate(target, subjectCertificate);
+    Path pemFile =
+        getPathFromResource("com/redhat/insights/tls/dummy.cert").resolveSibling("pem.cert");
+    Path keyFile =
+        getPathFromResource("com/redhat/insights/tls/dummy.key").resolveSibling("pem.key");
+    Files.write(pemFile, target.toArray());
+    Files.write(keyFile, generatedKeys.getPrivate().getEncoded());
+    new PEMSupport(new TestLogger(), new DefaultConfiguration()).createTLSContext(pemFile, keyFile);
+  }
+
+  @Test
+  public void testLoadingCertAndKeyOutOfDate() {
+    InsightsException exception =
+        Assertions.assertThrows(
+            InsightsException.class,
+            () ->
+                new PEMSupport(new TestLogger(), new DefaultConfiguration())
+                    .createTLSContext(
+                        getPathFromResource("com/redhat/insights/tls/dummy.cert"),
+                        getPathFromResource("com/redhat/insights/tls/dummy.key")));
+
+    Assertions.assertEquals("I4ASR0015: SSLContext creation error", exception.getMessage());
+    Assertions.assertTrue(exception.getCause() instanceof CertificateExpiredException);
+    Assertions.assertEquals(
+        "NotAfter: Sat Jan 14 15:24:00 CET 2023", exception.getCause().getMessage());
   }
 
```

Comment:
There should be explicit tests for a) time skew too large and b) Certificate revoked. Please add them.